lib/rules/no-return-await.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary `return await`",
+            category: "Best Practices",
+            recommended: false // TODO: set to true
+        },
+        fixable: false,
+        schema: [
+        ]
+    },
+
+    create(context) {
+        return {
+            ArrowFunctionExpression(node) {
+                if (node.async && node.body.type === "AwaitExpression") {
+                    const sourceCode = context.getSourceCode();
+
+                    context.report({
+                        node: sourceCode.getFirstToken(node.body),
+                        loc: node.body.loc,
+                        message: "await in an implicit-return async arrow function is redundant - just return the value directly.",
```

Comment:
I guess it seems to me that error messages should be as specific as possible - I could certainly make them all the same, but it seems like a missed opportunity to avoid them having to look up the documentation.